cards.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
```

Comment:
When you're writing code, you should be testing incrementally so you don't end up with code that has 20+ errors and just as many warnings. Looking through that many errors and warnings is painful, especially when grading. If you want real feedback on this (besides to also include plenty of comments, especially when you hand in non-working code), resubmit with more comments and some of these syntax errors fixed. I can't even tell a lot of your logic.